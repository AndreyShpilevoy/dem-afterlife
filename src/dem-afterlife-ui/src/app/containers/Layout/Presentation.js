import React from 'react';
import {string, node, shape} from 'prop-types';
import {sharedPropTypes} from 'utils';
import Notification from 'containers/Notification';
import Container from 'components/Container';
import Row from 'components/Row';
import Column from 'components/Column';
import {withStyles, ThemeProvider} from 'styles';
import Header from './components/Header';
import Footer from './components/Footer';
import globalStyles from './globalStyles';
import calculateStyles from './calculateStyles';

export const PresentationPure = ({children, themeName, navigationLinkArray, socialMediaLinkArray, styles}) => {
    const {contentWrapper, container, content} = styles;
    return <ThemeProvider name={themeName}>
        <div className={contentWrapper}>
            <Container className={container}>
                {/* title*/}
                <Notification />
                <Row>
                    <Header navigationLinkArray={navigationLinkArray}/>
                </Row>
                <Row className={content}>
                    <Column xs={12}>
                        {children}
                    </Column>
                </Row>
                <Row>
                    <Footer socialMediaLinkArray={socialMediaLinkArray}/>
                </Row>
            </Container>
        </div>
    </ThemeProvider>;
};

PresentationPure.propTypes = {
    styles: shape().isRequired,
    children: node,
    themeName: string,
    navigationLinkArray: sharedPropTypes.navigationLinkArray,
    socialMediaLinkArray: sharedPropTypes.socialMediaLinkArray
};

export default withStyles(theme => ({...globalStyles(), ...calculateStyles(theme)}))(PresentationPure);