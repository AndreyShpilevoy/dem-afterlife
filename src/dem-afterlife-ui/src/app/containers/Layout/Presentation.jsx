import React from 'react';
import {string, node, shape} from 'prop-types';
import {sharedPropTypes} from 'utils';
import Notification from 'containers/Notification';
import Breadcrumbs from 'containers/Breadcrumbs';
import Container from 'components/Container';
import Row from 'components/Row';
import Column from 'components/Column';
import {withStyles, ThemeProvider} from 'styles';
import Header from './components/Header';
import Footer from './components/Footer';
import globalStyles from './globalStyles';
import calculateStyles from './calculateStyles';

export const PresentationPure = ({children, themeName, navigationLinkArray, socialMediaLinkArray, breadcrumbArray, styles}) => {
    const {contentWrapper, container, content, breadcrumbs} = styles;
    return (<ThemeProvider name={themeName}>
        <div className={contentWrapper}>
            <Container className={container}>
                {/* title */}
                <Notification />
                <Row>
                    <Header navigationLinkArray={navigationLinkArray} />
                </Row>
                <Row className={breadcrumbs}>
                    <Column xs={12}>
                        <Breadcrumbs breadcrumbArray={breadcrumbArray} />
                    </Column>
                </Row>
                <Row className={content}>
                    <Column xs={12}>
                        {children}
                    </Column>
                </Row>
                <Row className={breadcrumbs}>
                    <Column xs={12}>
                        <Breadcrumbs breadcrumbArray={breadcrumbArray} />
                    </Column>
                </Row>
                <Row>
                    <Footer socialMediaLinkArray={socialMediaLinkArray} />
                </Row>
            </Container>
        </div>
    </ThemeProvider>);
};

PresentationPure.propTypes = {
    styles: shape().isRequired,
    children: node.isRequired,
    themeName: string,
    navigationLinkArray: sharedPropTypes.navigationLinkArray.isRequired,
    socialMediaLinkArray: sharedPropTypes.socialMediaLinkArray.isRequired,
    breadcrumbArray: sharedPropTypes.breadcrumbArray.isRequired
};

PresentationPure.defaultProps = {
    themeName: 'default'
};

export default withStyles(theme => ({...globalStyles(), ...calculateStyles(theme)}))(PresentationPure);