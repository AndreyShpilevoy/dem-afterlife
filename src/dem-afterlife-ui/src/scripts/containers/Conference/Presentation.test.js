/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0*/

import React from 'react';
import {shallow} from 'enzyme';
import Presentation from './Presentation';

jest.mock('styles/styler');

describe('Conference Presentation', () => {
    it('component match expected snapshot', () => {
        expect(shallow(<Presentation><div>{'Conference Presentation content'}</div></Presentation>)).toMatchSnapshot();
    });
});
