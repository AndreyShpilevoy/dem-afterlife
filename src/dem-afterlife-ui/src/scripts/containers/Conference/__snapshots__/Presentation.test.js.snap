// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conference Presentation component match expected snapshot 1`] = `
<div>
  <div>
    Conference Presentation content
  </div>
</div>
`;
